### ISSUE TITLE
Allow configuration of Volume plugins

### ISSUE THREAD

**Issue by witokondoria** [jenkinsci/docker-plugin#362]
Related to https://github.com/docker-java/docker-java/issues/329, the plugin should allow to configure a volume plugin

**Comment by KostyaSha** [comment]
What is plugin in your understanding?

**Comment by witokondoria** [comment]
Sorry for not being too clear: 

Docker added volume plugins, as defined in https://docs.docker.com/engine/extend/plugins_volume/, and mentioned in https://github.com/docker-java/docker-java/issues/329. Ideally a container hould be run defining a custom volume plugin, modifying its behaviour.

As docker-plugin uses docker-java to interact with a docker daemon (from my undestanding), here comes the related to link

**Comment by KostyaSha** [comment]
Right, but every plugin will have custom settings, could you provide any example of plugin that you really want to use/configure right now? (Or i didn't get something).

**Comment by witokondoria** [comment]
From my undertanding, a simple configuration equivalent to defining a --volume-driver docker parameter will be enough. Would be almost the same as defining volumes (a strings list).

Of course, each vloume plugin will have its own external configuration (as seen onhttps://github.com/calavera/docker-volume-glusterfs or https://github.com/gondor/docker-volume-netshare)

**Comment by magnayn** [comment]
The first step is probably to migrate the plugin to docker-java 3.x, as it
looks like the docker-java issue you link to will only be fixed in that
version.

I started to do this on the docker-java-next branch (it's 3.0.0-SNAPSHOT,
but now 3.0.0 is released it could be moved). There's some irritating
differences between 2.x and 3.x, but I doubt it is particularly hard to do.
I just ran out of time.

After that it'd probably be relatively simple to add the params on the job
plugin (there's a few PRs that have been merged that add small pieces like
this which is where I'd start).

**Comment by witokondoria** [comment]
Just so others are aware: docker-java was already bumped to 3.X as per #412

**Comment by bguerin** [comment]
The plugin now support the `mount` syntax as described here :
https://docs.docker.com/engine/reference/commandline/service_create/#add-bind-mounts-volumes-or-memory-filesystems

Does `volume-driver` option for named volume match your need ?

**Comment by witokondoria** [comment]
From a quick reading, I would say it provides the feature. im unfortunately not involved into using this plugin, so cant double check this statement.
Nevertheless, will close the issue with a big shoutout for the plugin mantainers
